archunit/src/test/java/com/tngtech/archunit/lang/syntax/elements/ClassesShouldTest.java

```diff
@@ -1767,6 +1770,26 @@ public void onlyCall_should_report_success_if_targets_are_non_resolvable(ArchRul
         assertThatRule(rule).checking(importClasses(classCallingUnresolvableTarget)).hasNoViolation();
     }
 
+    @Test
+    public void should_fail_on_empty_should_by_default() {
+        assertThatThrownBy(new ThrowingCallable() {
+            @Override
+            public void call() {
+                classes().that().areInterfaces().should().bePublic()
+                        .check(new ClassFileImporter().importClasses(getClass()));
+            }
+        }).isInstanceOf(AssertionError.class)
+                .hasMessageContaining("Rule failed to check any classes");
+    }
+
```

Comment:
I left them out, because we tested all three cases in `ArchRuleTest` and this here is more a sanity check for users of `ArchRule.Factory`, but maybe you're right :thinking: Can't harm I guess...

Reply:
My initial confusion was probably due to the fact that two parts change between those two tests and because I overlooked that the configuration sets `failOnEmptyShould` whereas the ArchRule has `allowEmptyShould`. 🙈